core/src/main/java/apoc/util/Utils.java

```diff
@@ -134,4 +134,40 @@ public byte[] compress(
         CompressionConfig conf = new CompressionConfig(config, CompressionAlgo.GZIP.name());
         return CompressionAlgo.valueOf(conf.getCompressionAlgo()).compress(data, conf.getCharset());
     }
+
+    private static String hexHash(final MessageDigest digest, final List<Object> values) {
+        for (final var value : values) digest.update(getBytesUtf8(toHashString(value)));
+        return Hex.encodeHexString(digest.digest());
+    }
+
+    /*
+     * This is not the most efficient way to produce a hash, but it is backwards compatible.
+     * This function is only intended to be used on strings (as documented on the hash functions above)
+     * But it turns out that is not how everyone is using it, so as a safety we have stable implementations
+     * for all neo4j types.
```

Comment:
Nice that you managed to keep backwards compatibility. Do you think we should mention something about the other types than strings in docs or not?

Reply:
Yes, perhaps we should point out that it is only meant for strings and other types can have bad performance?